jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala

```diff
@@ -552,12 +558,19 @@ class JmsConnectorsSpec extends JmsSpec with MockitoSugar {
         JmsProducerSettings(connectionFactory).withQueue("numbers")
       )
 
-      val completionFuture: Future[Done] = Source
-        .failed[JmsTextMessage](new RuntimeException("Simulated error"))
```

Comment:
This test was unreliable. Since setting the `jmsConnection` in `JmsConnector` is done through an async callback, the early arrival of the failure will cause the `jmsConnection` (which is not referenced) to not be closed. Need to cause the failure mid-stream instead of at the beginning for this test to be reliable.